client/zanata-client-commands/src/main/java/org/zanata/client/commands/PutUserCommand.java

```diff
@@ -21,22 +21,46 @@ public PutUserCommand(PutUserOptions opts) {
     }
 
     public void run() throws Exception {
-        Account account = new Account();
-        account.setEmail(getOpts().getUserEmail());
-        account.setName(getOpts().getUserName());
-        account.setUsername(getOpts().getUserUsername());
-        account.setPasswordHash(getOpts().getUserPasswordHash());
-        account.setApiKey(getOpts().getUserKey());
+
+        String userUsername = getOpts().getUserUsername();
+        String passwordHash = getOpts().getUserPasswordHash();
+        String userEmail = getOpts().getUserEmail();
+
+        Account account = getClientFactory().getAccountClient().get(userUsername);
+
+        if (account == null) {
+            log.info("Creating new account {}", userUsername);
+            account = new Account();
+            if (userEmail == null || userUsername == null) {
+                throw new RuntimeException("New user's name and email must be specified");
+            } else if (passwordHash == null || StringUtils.isBlank(passwordHash)) {
+                log.warn("Unusable passwordHash set, user will need to use the Forgot Password feature");
+                account.setPasswordHash("NULL");
```

Comment:
We shouldn't use the string "NULL" as a sentinel value[1]. If `Account.setPasswordHash` won't accept `null`, we should probably change it, but in the meantime it would be better to pass an empty string.

[1] https://www.wired.com/2015/11/null/